### PR TITLE
APIGOV-21297 - Watch for Agent resource

### PR DESCRIPTION
**Pull request by vivekschauhan**
- Updates to watch for update events on agent resource
- Includes refactoring to manage agent resource

### CODE REVIEW

File: pkg/agent/handler/handlers.go [hunk 1]
```diff
@@ -107,3 +117,29 @@ func (c categoryHandler) handle(action proto.Event_Type, resource *apiv1.Resourc
...
+func (h *agentResourceHandler) Handle(action proto.Event_Type, resource *v1.ResourceInstance) error {
+	if h.agentResourceManager != nil && action == proto.Event_UPDATED {
+		kind := resource.Kind
+		switch kind {
+		case discoveryAgent:
+			fallthrough
+		case traceabilityAgent:
+			fallthrough
+		case governanceAgent:
+			h.agentResourceManager.SetAgentResource(resource)
```
Comment (jcollins-axway) [inline pkg/agent/handler/handlers.go:141]: Do other resource kinds come in here?  Is this switch necessary?

**Review by jcollins-axway — APPROVED** [review]

File: pkg/agent/handler/handlers.go (hunk 1 above)
Comment (vivekschauhan) [reply pkg/agent/handler/handlers.go:141]: The handler gets invoked for all resource events received by the watch client, so this handler does the check for agent resource kind to process them further

File: pkg/agent/resource/manager.go [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+	case v1alpha1.TraceabilityAgentResourceName:
+		agentRes := createTraceabilityAgentStatusResource(a.cfg.GetAgentName(), status, prevStatus, message)
+		resourceInstance, _ := agentRes.AsInstance()
+		return resourceInstance
+	case v1alpha1.GovernanceAgentResourceName:
+		agentRes := createGovernanceAgentStatusResource(a.cfg.GetAgentName(), status, prevStatus, message)
+		resourceInstance, _ := agentRes.AsInstance()
+		return resourceInstance
+	default:
+		panic(ErrUnsupportedAgentType)
```
Comment (tsjohns9) [inline pkg/agent/resource/manager.go:212]: Can we return an error instead of a panic?

**Review by tsjohns9 — APPROVED** [review]

File: pkg/agent/resource/manager.go (hunk 2 above)
Comment (vivekschauhan) [reply pkg/agent/resource/manager.go]: Done

**Pull request merged**